### PR TITLE
Activity

### PR DESCRIPTION
**Pull request by FatimaMubeenSh**
New: Activity migrations, test cases, model, controller.
Fix: Methods to get surveys and packages added to models instead of fetching them directly in home controller.

### CODE REVIEW

File: app/assets/javascripts/home.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (zohaibbutt13) [inline app/assets/javascripts/home.coffee]: remove files we are not using yet

Comment (FatimaMubeenSh) [reply app/assets/javascripts/home.coffee]: done

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+header.master-head {
+  position: relative;  
+  background: url("bg-masterhead.jpg") no-repeat center center;
+  background-size: cover;
+  padding-top: 8rem;
+  padding-bottom: 8rem;
+}
+
```
Comment (aliahmad7) [inline app/assets/stylesheets/custom.scss]: remove extra spaces

File: app/controllers/activity_controller.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'will_paginate/array'
+
+class ActivityController < ApplicationController
+  # get activity/index
+  def index
+    # if @current_user.role == 'admin'
```
Comment (aliahmad7) [inline app/controllers/activity_controller.rb]: remove this commented code if no needed

File: app/controllers/home_controller.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require 'will_paginate/array'
+
+
+class HomeController < ApplicationController
+  # get home/index
+  def index
+    @surveys = Survey.get_public_surveys.paginate(page: params[:page], per_page: 6)
```
Comment (aliahmad7) [inline app/controllers/home_controller.rb]: remove page: and per_page:

File: db/migrate/20190718124955_create_activities.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateActivities < ActiveRecord::Migration
+  def change
+    create_table :activities do |t|
+      t.references :company
```
Comment (aliahmad7) [inline db/migrate/20190718124955_create_activities.rb]: t.integer = company_id, index: true ...can use this?

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -12,5 +12,4 @@
 //
 //= require jquery
 //= require jquery_ujs
-//= require turbolinks
```
Comment (aliahmad7) [inline app/assets/javascripts/application.js]: remove turbolinks

Comment (FatimaMubeenSh) [reply app/assets/javascripts/application.js]: The highlighted red line indicates it's been removed already.

File: db/migrate/20190718124955_create_activities.rb (hunk 5 above)
Comment (FatimaMubeenSh) [reply db/migrate/20190718124955_create_activities.rb]: Done

File: app/controllers/home_controller.rb (hunk 4 above)
Comment (FatimaMubeenSh) [reply app/controllers/home_controller.rb]: done

File: app/controllers/activity_controller.rb (hunk 3 above)
Comment (FatimaMubeenSh) [reply app/controllers/activity_controller.rb]: done

File: app/assets/stylesheets/custom.scss (hunk 2 above)
Comment (FatimaMubeenSh) [reply app/assets/stylesheets/custom.scss]: done

File: app/controllers/activity_controller.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+require 'will_paginate/array'
+
+class ActivityController < ApplicationController
+  # get activity/index
+  def index
+     @activities = Activity.get_user_activities(@current_user)
```
Comment (zohaibbutt13) [inline app/controllers/activity_controller.rb]: remove extra space

File: app/models/activity.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
+class Activity < ActiveRecord::Base
+  belongs_to :company
+  belongs_to :trackable, :polymorphic => true
```
Comment (zohaibbutt13) [inline app/models/activity.rb]: new syntax

File: app/models/activity.rb [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  belongs_to :owner, class_name: 'User'
+
+  validates_presence_of :action, :company_id, :owner_id,
+                        :trackable_id, :trackable_type
+
+  scope :user_activities, ->(user_id) { where("owner_id = ?", user_id) }
+  scope :company_activities, ->(company_id) { where("company_id = ?", company_id) }
+
+  def self.get_user_activities(user)
+    user ||= { id: 3, role: 'admin', company_id: 1 }
```
Comment (zohaibbutt13) [inline app/models/activity.rb]: why need this?

File: app/models/activity.rb [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    if user[:role] == 'member' || user[:role] == 'supervisor'
```
Comment (zohaibbutt13) [inline app/models/activity.rb]: then what's the difference between supervisor and member

File: app/models/company.rb [hunk 11]
```diff
@@ -9,6 +9,7 @@ class Company < ActiveRecord::Base
   has_many :options
   has_many :responses
   has_many :user_settings
+  has_many :activities
```
Comment (zohaibbutt13) [inline app/models/company.rb]: please add dependent: :destroy on all associations in company

File: app/models/concerns/trackable.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+module Trackable
```
Comment (zohaibbutt13) [inline app/models/concerns/trackable.rb]: add line in each model instead of making concern

File: app/controllers/activity_controller.rb (hunk 7 above)
Comment (FatimaMubeenSh) [reply app/controllers/activity_controller.rb]: done

File: app/models/activity.rb (hunk 8 above)
Comment (FatimaMubeenSh) [reply app/models/activity.rb]: done

File: app/models/activity.rb (hunk 9 above)
Comment (FatimaMubeenSh) [reply app/models/activity.rb]: resolved

File: app/models/activity.rb (hunk 10 above)
Comment (FatimaMubeenSh) [reply app/models/activity.rb]: discussed

File: app/models/company.rb (hunk 11 above)
Comment (FatimaMubeenSh) [reply app/models/company.rb]: done

File: app/models/concerns/trackable.rb (hunk 12 above)
Comment (FatimaMubeenSh) [reply app/models/concerns/trackable.rb]: done

**Pull request merged**